sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -401,6 +401,13 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     }
   }
 
+  override def alterTableStats(tableDefinition: CatalogTable): Unit = withClient {
```

Comment:
Yeah, agree with @cloud-fan . 

If we want to set it using `alter table`, we should use the dedicated command (just like what Hive does):
```SQL
ALTER TABLE UPDATE STATISTICS SET
```
Let us remove `alterTableStats` and minimize the code changes? We can discuss how to do it properly when we start this JIRA: https://issues.apache.org/jira/browse/SPARK-17282